### DISCUSSION TITLE
After 12.2.0 update Middleware File Issue Facing while deployment

### DISCUSSION THREAD

**Discussion by Vrund9** [vercel/next.js#38200] [Help]
Hello,

I Write my middleware code in pages/_middleware.js but after update facing issue in production deployment then i find some solution and update nextjs then move and rename the file to middleware.js and now I am facing issue my App that issue is first time page goes for error and after refresh the page will display. so how I can solve this.

My Middleware file code is:

```
import { NextResponse } from 'next/server'

export function middleware(req) {
  const amp = req.nextUrl.searchParams.get('amp')
  // strictly check that amp is empty string
  if (amp === '') {
    req.nextUrl.searchParams.set('amp', '1')
    return NextResponse.rewrite(req.nextUrl)
  }

  return NextResponse.next()
}
```

Thanks in Advance

**Comment by icyJoseph** [comment]
What error are you seeing?

Just tested this with 12.2.0, and it works. Using 'hybrid' mode for the pages.

```
.
├── README.md
├── components
├── hooks
├── jest.config.js
├── middleware.ts
├── next-env.d.ts
├── next.config.js
├── node_modules
├── package.json
├── pages
├── public
├── tsconfig.json
└── yarn.lock
```

Notice that middleware is now at the top, sibling to `pages`.
